### DISCUSSION TITLE
Native- I mean External file picker

### DISCUSSION THREAD

**Discussion by Speykious** [ppy/osu#17480] [Ideas] [answered]
I noticed that osu!lazer has its own integrated file picker. Why do that instead of using an operating system's most widespread file picker?
Is it inconvenient that, when on fullscreen, osu! is put in the background for the file picker to take its place?
Or is it simply that there isn't any viable solution for C# on .NET?
Or is it something else, idk, aesthetics?

**Comment by bdach** [comment]
linux and android don't even have the proper notion of a "native file picker", for one. on those OSes what is the "file picker" depends on what you have installed on the OS.

**Reply by frenzibyte** [reply]
Also relevant: 
 - https://github.com/ppy/osu/discussions/17261

**Reply by Speykious** [reply]
Indeed, sorry for the poor choice of words. As I'm more familiar with Linux than anything else, I know for instance that there are options such as Zenity, or GTK2/3 file pickers, and various other programs. If the situation is the same on Android, then I guess I should have asked, why make osu! implement its own file pickers in-game instead of supporting the most widespread file pickers for each OS, whether it be native or just a common program to its ecosystem?

**Comment by peppy** [comment, marked as answer]
You answered this in your question. They can't be displayed when running exclusive fullscreen.
